src/shared/machine-id-setup.c

```diff
@@ -54,7 +54,14 @@ static int generate_machine_id(const char *root, sd_id128_t *ret) {
 
         assert(ret);
 
-        /* First, try reading the D-Bus machine id, unless it is a symlink */
+        /* First, try reading the machine ID from /run/machine-id, which may not be mounted on
+         * /etc/machine-id yet. This is important on switching root, Otherwise, machine ID may be changed
+         * after the transition. */
+        if (empty_or_root(root) && running_in_chroot() <= 0 &&
+            id128_read("/run/machine-id", ID128_FORMAT_PLAIN, ret) >= 0)
+                return 0;
```

Comment:
Hmm, this is our own file, so placing this bit of logic in `generate_machine_id` seems wrong? And this might result in reading from `/run/machine-id` and later writing the same id into it again.

Reply:
> Hmm, this is our own file, so placing this bit of logic in `generate_machine_id` seems wrong?

OK. Renamed `generate_machine_id()` -> `acquire_machine_id()`.

> And this might result in reading from /run/machine-id and later writing the same id into it again.

That should not be a big matter, I think.

Reply:
> That should not be a big matter, I think.

Yes, but I'd still like to avoid it, as it might be confusing for others. Maybe make acquire__machine_id return 1 to indicate the case?

Reply:
OK. Applied the suggestion.